doc/hooks-config.rst

```diff
@@ -0,0 +1,127 @@
+                        "bar": 42,
+                        "enable_x": True,
+                    },
+                    "another_hook_id": {
+                        "baz": "value",
+                    }
+                },
+                ...)
+
+
+.. _supported hooks and options:
+
+Supported hooks and options
+===========================
+
+This section lists hooks that implement support for configuration
+options. For each hook (or group of hooks), we provide the `hook
+identifier` and the list of supported options.
+
+GObject introspection (gi) hooks
+--------------------------------
+
+The options passed under `gi` hook identifier control the collection
+of GLib/Gtk resources (themes, icons, translations) in various
+hooks related to ``GObject introspection`` (i.e., ``hook-gi.*``).
+
+They are especially useful when freezing ``Gtk``-based applications on
+linux to limit the amount of themes and icons collected from the system
+``/usr/share`` directory.
+
+**Hook identifier:** ``gi``
+
+**Options**
+
+ * ``languages`` [*list of strings*]: list of locales (e.g., ˙en_US˙) for
+   which translations should be collected. By default, ``gi`` hooks
+   collect all available translations.
+
+ * ``icons`` [*list of strings*]: list of icon themes (e.g., `Adwaita`)
+   that shoud be collected. By default, ``gi`` hooks collect all available
+   icon themes.
+
+ * ``themes`` [*list of strings*]: list of Gtk themes (e.g., `Adwaita`)
+   that shoud be collected. By default, ``gi`` hooks collect all available
+   icon themes.
+
+**Example**
+
+Collect only ``Adwaita`` theme and icons, and limit the collected
+translations to British English and Simplified Chinese:
+
+.. code-block:: python
+
+    a = Analysis(["my-gtk-app.py"],
+                 ...,
+                 hooksconfig={
+                     "gi": {
+                         "icons": ["Adwaita"],
+                         "themes": ["Adwaita"],
+                         "languages": ["en_GB", "zh_CN"]
+                     }
+                 },
+                 ...)
+
+
+
+
+Adding an option to the hook
+============================
+
+Implementing support for hook options requires access to ``hook_api``
+object, which is available only when hook implements the ``hook(hook_api)``
+function (as described :ref:`here <The hook(hook_api) Function>`).
+
+The value of a hook's configuration option can be obtained using the
+:func:`PyInstaller.utils.hooks.get_hook_config` function:
+
+.. code-block:: python
+
+    # hook-mypackage.py
+    from PyInstaller.utils.hooks import get_hook_config
+
+    # Processing unrelated to hook options, using global hook values
+    binaries, datas, hiddenimports = ...
+
+    # Collect extra data
+    def hook(hook_api):
+        # Boolean option 'collect_extra_data'
+        if get_hook_config(hook_api, 'myhook', 'collect_extra_data'):
+            extra_datas = ...  # Collect extra data
+            hook_api.add_datas(extra_datas)
+
+
```

Comment:
Can we squeeze in an example setting the corresponding spec file option here too?

Reply:
Done.